FizzBuzz/lib/FizzBuzzImproved.rb

```diff
@@ -0,0 +1,18 @@
+class FizzBuzzImproved
+  def initialize
+    @set = {}
+    @val = ""
+  end
+
+  def set(arbitrary_number, display_value)
+    @set[arbitrary_number] = display_value
+  end
+
+  def final(num_in)
+    @set.each do |num, dis|
```

Comment:
I would change dis to a more descriptive variable.  display, display_string, etc.